src/org/labkey/test/tests/SampleTypeNameExpressionTest.java

```diff
@@ -232,4 +286,298 @@ private void verifyNames(String sampleTypeName, String header, String nameExpres
+
+        createPage.setNameExpression(nameExpression);
+
+        createPage.addFields(Arrays.asList(
+                new FieldDefinition("Int", FieldDefinition.ColumnType.Integer),
+                new FieldDefinition("Str", FieldDefinition.ColumnType.String)));
+
+        createPage.clickSave();
+
+        log(String.format("Go to the 'overview' page for sample '%s' in sample type '%s'", PARENT_SAMPLE_01, PARENT_SAMPLE_TYPE));
+        Long sampleRowNum = SampleTypeAPIHelper.getSampleIdFromName(getProjectName(), PARENT_SAMPLE_TYPE, Arrays.asList(PARENT_SAMPLE_01)).get(PARENT_SAMPLE_01);
+
+        String url = WebTestHelper.buildRelativeUrl("experiment", getCurrentContainerPath(), "showMaterial", Map.of("rowId", sampleRowNum));
+        beginAt(url);
+
+        log("Derive a sample from this sample but give it no name. The name expression should be used to name the derived sample.");
+
+        waitForElement(Locator.linkWithText("derive samples from this sample"));
+
+        clickAndWait(Locator.linkWithText("derive samples from this sample"));
+
+        selectOptionByText(Locator.name("targetSampleTypeId"),  String.format("%s in /%s", sampleType, getProjectName()));
+        clickButton("Next");
+
+        String flagString = "Hello, I'm a derived sample.";
+        setFormElement(Locator.name("outputSample1_Int"), "987");
+        setFormElement(Locator.name("outputSample1_Str"), flagString);
+        clickButton("Submit");
+
+        waitForElement(Locator.tagWithText("td", flagString));
+
+        String derivedSampleName = Locator.tagWithText("td", "Name:").followingSibling("td").findElement(getDriver()).getText();
+
+        checker().verifyTrue("Name of derived sample doesn't look correct.", derivedSampleName.contains("Parent Sample"));
+
+        checker().verifyTrue("Doesn't look like there is a link to the parent sample.", isElementPresent(Locator.linkWithText(PARENT_SAMPLE_01)));
+
+    }
+
+    /**
+     * Simple helper to build the expected text in the tool-tip for the name expression.
+     *
+     * @param expectedPreview What the expected preview example should look like. If null will return default tool-tip.
+     * @param error Should the name expression generate an error?
+     * @return The expected text in the tool-tip including header and other text.
+     */
+    private String generateExpectedToolTip(@Nullable String expectedPreview, boolean error)
+    {
+
+        StringBuilder expectedToolTip = new StringBuilder();
+
+        expectedToolTip.append("Naming Pattern\n");
+        expectedToolTip.append("Pattern used for generating unique IDs for this sample type.\n");
+
+        if(expectedPreview != null)
+        {
+            if(!error)
+            {
+                expectedToolTip.append("Example of name that will be generated from the current pattern: ");
+            }
+
+            expectedToolTip.append(expectedPreview);
+            expectedToolTip.append("\n");
+        }
```

Comment:
if there is error, wouldn't there be "Unable to generate example name from the current pattern. Check for syntax errors." msg?

Reply:
Yes it does.
In that case the error message is the value sent in. Example:
`String expectedMsg = generateExpectedToolTip("Unable to generate example name from the current pattern. Check for syntax errors.", true);`
The testErrorsAndWarning test has more examples.
I will update the comment to clarify.

Reply:
hmm, is there any reason to not have the function check for the error string without having to pass as param? I think this is a static msg that doesn't change based on error reason?